presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -109,6 +112,7 @@
     private final Map<QueryId, Long> preAllocations = new HashMap<>();
     private final Map<QueryId, Long> preAllocationsConsumed = new HashMap<>();
 
+    @GuardedBy("this")
```

Comment:
Are you concerned about contention on this lock?

Reply:
This field was already accessed by holding this lock (I added the annotation and added synchronized to some methods which were called from already synchronized methods), the additional locking will come from the HTTP threads when the endpoint is called, which should not cause any issues.